File: frontend/src/api/demoReadiness.ts
// Safe Paper demo preflight. The endpoint accepts no credentials and its only
// broker request is GET /v2/account; it cannot create, modify or cancel an
// order.

import { apiGet, apiPost } from "./client";

export type PaperConnectionStatus = "NOT_CONFIGURED" | "NOT_RUN" | "VERIFIED" | "AUTH_FAILED" | "UNREACHABLE";
export type McpSessionStatus = "NOT_STARTED" | "STARTING" | "HEALTHY" | "RECONNECTING" | "STOPPED" | "UNKNOWN";
export type KillSwitchReadinessStatus = "DISENGAGED" | "ENGAGED" | "UNKNOWN";

export type PaperDemoReadiness = {
  account_configured: boolean;
  account_connected: boolean;
  paper_url_locked: boolean;
  paper_connection_status: PaperConnectionStatus;
  paper_connection_checked_at: string | null;
  mcp_session_status: McpSessionStatus;
  active_option_contract_count: number;
  options_last_synced_at: string | null;
  trading_kill_switch_status: KillSwitchReadinessStatus;
  ready_for_paper_demo: boolean;
  non_transactional: true;
};

export async function fetchPaperDemoReadiness(): Promise<PaperDemoReadiness> {
  return apiGet<PaperDemoReadiness>("/api/demo-readiness");
}

export async function runPaperPreflight(): Promise<PaperDemoReadiness> {
  return apiPost<PaperDemoReadiness>("/api/demo-readiness/paper-preflight");
}

File: backend/app/schemas/demo_readiness.py
"""Read-only Paper demo readiness contracts.

The status is intentionally separate from the trading/order contracts: it is
only a checklist and a non-transactional broker connectivity probe.  It never
accepts credentials and it cannot create, modify, or cancel an order.
"""

from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel


PaperConnectionStatus = Literal[
    "NOT_CONFIGURED",
    "NOT_RUN",
    "VERIFIED",
    "AUTH_FAILED",
    "UNREACHABLE",
]
McpReadinessStatus = Literal["NOT_STARTED", "STARTING", "HEALTHY", "RECONNECTING", "STOPPED", "UNKNOWN"]
KillSwitchReadinessStatus = Literal["DISENGAGED", "ENGAGED", "UNKNOWN"]


class PaperDemoReadinessOut(BaseModel):
    """Sanitised, authenticated readiness for the Paper options demo."""

    account_configured: bool
    account_connected: bool
    paper_url_locked: bool
    paper_connection_status: PaperConnectionStatus
    paper_connection_checked_at: datetime | None
    mcp_session_status: McpReadinessStatus
    active_option_contract_count: int
    options_last_synced_at: datetime | None
    trading_kill_switch_status: KillSwitchReadinessStatus
    ready_for_paper_demo: bool
    non_transactional: bool = True
